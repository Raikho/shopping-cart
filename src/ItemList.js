// import { useOutletContext } from 'react-router-dom';

const ItemList = props => {

    // const [items] =  useOutletContext();
    const items = props.items;

    return (
        <div className="item-list">
            {items.map(item => (
                <div className='item' key={item.id}>
                    <div className="name">{item.name}</div>
                    <img src={require(`./assets/${item.img}.jpg`)} alt={item.img} />
                    <div className="price">{`$${item.price}.00`}</div>
                    <button onClick={() => props.handleAdd(item.id)}>Buy</button>
                </div>
            ))}
        </div>
    )
}

export default ItemList;