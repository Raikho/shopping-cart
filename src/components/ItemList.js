import { useParams } from 'react-router-dom';

const ItemList = props => {
    const { getItems, handleAddItem } = props;

    const items = getItems(useParams().cat);

    return (
        <div className="item-list">
            {items.map(item => (
                <div className='item' key={item.id}>
                    <div className="name">{item.name}</div>
                    <img src={require(`../assets/${item.img}.jpg`)} alt={item.img} />
                    <div className="price">{`$${item.price}.00`}</div>
                    <button onClick={() => handleAddItem(item.id)}>Buy</button>
                </div>
            ))}
        </div>
    )
}

export default ItemList;