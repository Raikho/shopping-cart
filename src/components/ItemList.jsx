import Item from './Item'

export default function ItemList({ items, onChangeAmount }) {
    return (
        <div className="item-list">
            ItemList
            {items.map(item => (
                <Item 
                    key={item.id}
                    item={item}
                    onChangeAmount={onChangeAmount}
                />
            ))}
        </div>
    );
}