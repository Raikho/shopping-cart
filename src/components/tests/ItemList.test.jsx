import { describe, it, expect, vi } from 'vitest';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import ItemList from '../ItemList.jsx';

vi.mock('../Item.jsx', () => ({
    default: ({ item, onChangeAmount }) => 
        <>
            <div>{item.name}</div>
            <button onClick={onChangeAmount}>{`add${item.name}`}</button>
        </> 
}));
const mockFn = vi.fn();
const mockItems = [{name: 'Betta', id: 1}, {name: 'Oscar', id: 2}];

describe('ItemList component', () => {
    it('renders', () => {
        const { container } = render(<ItemList 
            items={mockItems} onChangeAmount={mockFn} />);
        expect(container).toMatchSnapshot();
    });

    it('renders each item', () => {
        render(<ItemList items={mockItems} onChangeAmount={mockFn} />);
        expect(screen.queryByText('Betta')).toBeInTheDocument();
        expect(screen.queryByText('Oscar')).toBeInTheDocument();
    });

    it('passes along amount function', async() => {
        render(<ItemList items={mockItems} onChangeAmount={mockFn} />);
        const user = userEvent.setup();
        const button = screen.getByRole('button', {name: 'addBetta'});
        await user.click(button);
        expect(mockFn).toHaveBeenCalled();
    });
});
