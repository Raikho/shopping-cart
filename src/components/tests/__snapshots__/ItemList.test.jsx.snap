// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ItemList component > renders 1`] = `
<div>
  <div
    class="item-list"
  >
    <div>
      Betta
    </div>
    <button>
      addBetta
    </button>
    <div>
      Oscar
    </div>
    <button>
      addOscar
    </button>
  </div>
</div>
`;
